services/inputhost/exthost.go

```diff
@@ -179,25 +189,29 @@ var (
 
 func newExtConnection(destUUID string, pathCache *inPathCache, extUUID string, numReplicas int, loadReporterFactory common.LoadReporterDaemonFactory, logger bark.Logger, tClients common.ClientFactory, shutdownWG *sync.WaitGroup, limitsEnabled bool) *extHost {
 	conn := &extHost{
-		streams:                 make(map[storeHostPort]*replicaInfo),
-		extUUID:                 extUUID,
-		destUUID:                destUUID,
-		destType:                pathCache.destType,
-		logger:                  logger.WithField(common.TagModule, `extHost`),
-		tClients:                tClients,
-		lastSuccessSeqNo:        int64(-1),
-		lastSuccessSeqNoCh:      nil,
-		notifyExtCacheClosedCh:  pathCache.notifyExtHostCloseCh,
-		notifyExtCacheUnloadCh:  pathCache.notifyExtHostUnloadCh,
-		putMessagesCh:           pathCache.putMsgCh,
-		replyClientCh:           make(chan writeResponse, defaultBufferSize),
-		closeChannel:            make(chan struct{}),
-		streamClosedChannel:     make(chan struct{}),
-		numReplicas:             numReplicas,
-		shutdownWG:              shutdownWG,
-		forceUnloadCh:           make(chan struct{}),
-		limitsEnabled:           limitsEnabled,
-		maxSequenceNumber:       common.GetRandInt64(int64(extentRolloverSeqnumMin), int64(extentRolloverSeqnumMax)),
+		streams:                make(map[storeHostPort]*replicaInfo),
+		extUUID:                extUUID,
+		destUUID:               destUUID,
+		destType:               pathCache.destType,
+		logger:                 logger.WithField(common.TagModule, `extHost`),
+		tClients:               tClients,
+		lastSuccessSeqNo:       int64(-1),
+		lastSuccessSeqNoCh:     nil,
+		notifyExtCacheClosedCh: pathCache.notifyExtHostCloseCh,
+		notifyExtCacheUnloadCh: pathCache.notifyExtHostUnloadCh,
+		putMessagesCh:          pathCache.putMsgCh,
+		replyClientCh:          make(chan writeResponse, defaultBufferSize),
+		closeChannel:           make(chan struct{}),
+		streamClosedChannel:    make(chan struct{}),
+		numReplicas:            numReplicas,
+		shutdownWG:             shutdownWG,
+		forceUnloadCh:          make(chan struct{}),
+		limitsEnabled:          limitsEnabled,
+		maxSequenceNumber:      common.GetRandInt64(int64(extentRolloverSeqnumMin), int64(extentRolloverSeqnumMax)),
+		// by default, max size is now based on the max allowed sequence number
+		// and an average message size of 1k;
+		// So in general, it will be max of 20GB((or 20000MB)), assuming max seq num is 20 million (or 20000MB)
+		maxSizeMB:               float32(extentRolloverSeqnumMax*1024) / bytesPerMB,
```

Comment:
i actually had it 20G because we have 2T HDD (most of our prod machines are as such) and so 1% of this ~20G. But I agree we can make this randomized based on the sealSeqNumber..